File: solns/clojure/53.clj
(load-file "lib/number-theory.clj")
(require
  '[lib.number-theory :refer [choose]])


(def result
  (count
    (for [n (range 1 (inc 100))
          r (range 1 (dec n))
          :let [c (choose n r)]
          :when (> c 1000000)]
        c)))

(println result)
(assert (= result 4075))

File: solns/53.py
# Problem 53: Combinatoric Selections.
#
# Computes the number distinct values of (n choose r) for 1 ≤ n ≤ limit that are greater than 1_000_000.
#
# See {@link https://projecteuler.net/problem=53}.
import math


def run(limit: int) -> int:
    count = 0
    for n in range(1, limit + 1):
        for r in range(1, n):
            c = math.comb(n, r)
            if c > 1_000_000:
                count += 1
    return count


def test_run():
    assert run(100) == 4075
